daml-assistant/src/DA/Daml/Assistant/Env.hs

```diff
@@ -144,6 +147,22 @@ getDamlPath = wrapErr "Determining daml home directory." $ do
         hasDamlConfig :: FilePath -> IO Bool
         hasDamlConfig p = doesFileExist (p </> damlConfigName)
 
+-- | Get the DAML cache folder. This defaults to $XDG_CACHE_HOME/daml.
+getCachePath :: IO CachePath
+getCachePath =
+    wrapErr "Determing daml cache directory." $ do
+        overrideWithEnvVar damlCacheEnvVar makeAbsolute CachePath $ do
+            errOrCacheDir <- try @IO @IOException $ do
+              -- getXdgDirectory throws an exception when no XDG_CACHE_HOME environment is set and
+              -- no HOME directory is set. In this case we fall back to the daml path.
+              cacheDir <- getXdgDirectory XdgCache "daml"
```

Comment:
looking through the code I’m actuall a bit confusey. Which part is failing if `$HOME` is not set? Afaict, both `getAppUserDirectory` and `getXdgDirectory` end up calling [getHomeDirectoryInternal](https://hackage.haskell.org/package/directory-1.3.6.1/docs/src/System.Directory.Internal.Posix.html#getHomeDirectoryInternal) which handles HOME not being set the same way.

Reply:
ah https://github.com/haskell/directory/issues/102, I think you can upgrade directory and sidestep this whole mess.

Reply:
yes please!